### PR TITLE
docs: updates

### PR DESCRIPTION
**Pull request by mariux**

### CODE REVIEW

File: docs/cli/orchestration/order-of-execution.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: Order of Execution
+description: Define the order stacks are executed in.
```
Comment (soerenmartius) [inline docs/cli/orchestration/order-of-execution.md:3]:
```suggestion
description: Learn about the order of execution when orchestration stacks with Terramate.
```

File: docs/cli/orchestration/order-of-execution.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+outline: [2, 3]
```
Comment (soerenmartius) [inline docs/cli/orchestration/order-of-execution.md:4]: Shall we add this as default settings to the config? https://vitepress.dev/reference/default-theme-config#outline

File: docs/cli/orchestration/order-of-execution.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+# Order of Execution
+
+## Implicit Order of Execution
+
+implicit order will be detected by Terramate depending on integrations and hierarchy of stacks.
```
Comment (soerenmartius) [inline docs/cli/orchestration/order-of-execution.md:11]:
```suggestion
Implicit order will be detected by Terramate depending on integrations and hierarchy of stacks.
```

File: docs/cli/orchestration/order-of-execution.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+### Parent and Child Stacks
+
+Currently Terramate orders stacks implicitly when they are nested.
+Nested Stacks are stacks that are in a subdirectory of stack.
```
Comment (soerenmartius) [inline docs/cli/orchestration/order-of-execution.md:16]:
```suggestion
nested stacks are stacks that are in a subdirectory of stack.
```

File: docs/cli/orchestration/run-commands-in-stacks.md [hunk 5]
```diff
@@ -1,122 +1,112 @@
...
+terramate run --parallel 100 -- echo "hello from stack in parallel"
 ```
 
-::: info
-The default base ref is `origin/main` when working in a feature branch and
-`HEAD^` when on main. The base ref can be changed in the project configuration
-(or with `-B`), but the defaults allow for the most common workflow where all
-changed stacks in a feature branch should be previewed in a PR and applied on merge.
+::: warning
+It is not possible to run `terraform init` in parallel when provider caching is enabled via `TF_PLUGIN_CACHE_DIR` as terraform does not support this mode of operation at this time.
```
Comment (soerenmartius) [inline docs/cli/orchestration/run-commands-in-stacks.md:93]:
```suggestion
It is not possible to run `terraform init` in parallel when provider caching is enabled via `TF_PLUGIN_CACHE_DIR` as Terraform does not support this mode of operation at this time.
```

File: docs/cli/orchestration/run-commands-in-stacks.md [hunk 6]
```diff
@@ -1,122 +1,112 @@
...
 ```
 
-### Changing the run scope and order
+This is useful when running destructive operations where dependent stacks need to remove their configuration before other stacks.
 
-Sometimes, you want to change the [default order of execution](./index.md#order-of-execution), which can be done
-with the [`before`](../stacks/configuration.md#stackbefore-setstringoptional) and
-[`after`](../stacks/configuration.md#stackafter-setstringoptional) attributes in the configuration of a stack.
-For details, please see the [stacks configuration](../stacks/configuration.md#configuring-the-order-of-execution) documentation.
+A very good example here is the destroy operation of Terraform to destroy stacks in opposite order:
```
Comment (soerenmartius) [inline docs/cli/orchestration/run-commands-in-stacks.md:108]:
```suggestion
An example use-case is the destroy operation of Terraform to destroy stacks in opposite order:
```

File: docs/cli/stacks/configuration.md [hunk 7]
```diff
@@ -1,143 +1,178 @@
 ---
 title: Stack Configuration
 description: Learn how to configure the metadata and orchestration behavior of stacks in Terramate.
+outline: [2, 3]
 ---
 
 # Configure stacks
 
-Stacks can be configured in the `stack {}` block (by default in `stack.tm.hcl`). The options available cover:
+Terramate detects stacks based on the exitance of a `stack {}` Block. The name of the file is not important and can be different from `stack.tm.hcl`. There can be exactly one stack block defined in a stack.
```
Comment (soerenmartius) [inline docs/cli/stacks/configuration.md:9]:
```suggestion
Terramate detects stacks based on the existence of a `stack {}` block. The name of the file is not important and can be different from `stack.tm.hcl`. There can be exactly one stack block defined in a stack.
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

**Comment by soerenmartius** [comment]
We usually don't do any uppercase in headlines other than brands (e.g. `Terraform`) or so but it's not super important now

File: docs/cli/orchestration/order-of-execution.md (hunk 2 above)
Comment (mariux) [reply docs/cli/orchestration/order-of-execution.md:4]: does not matter fmpov.

**Review by soerenmartius — APPROVED** [review]

**Pull request merged**